wx/lib/agw/speedmeter.py

```diff
@@ -324,10 +324,23 @@ def __init__(self, parent, id=wx.ID_ANY, pos=wx.DefaultPosition, size=wx.Default
         self.Bind(wx.EVT_SIZE, self.OnSize)
         self.Bind(wx.EVT_ERASE_BACKGROUND, lambda x: None)
 
-        # OnSize called to make sure the buffer is initialized.
-        # This might result in OnSize getting called twice on some
-        # platforms at initialization, but little harm done.
-        self.OnSize(None)
+        self._isWindowCreated = False
+        if '__WXGTK__' in wx.PlatformInfo:
+            self.Bind(wx.EVT_WINDOW_CREATE, self.doOnSize)
+        else:
+            # OnSize called to make sure the buffer is initialized.
+            # This might result in OnSize getting called twice on some
+            # platforms at initialization, but little harm done.
+            self._isWindowCreated = True
+            self.OnSize(None)
+
+
+    def doOnSize(self, evt):
+        """
+        Method to call OnSize on GTK when window is created.
+        """
+        self._isWindowCreated = True
+        self.OnSize(evt)
```

Comment:
The `evt` object here will be the wrong type for an `EVT_SIZE` event.  In this case is doesn't matter since it is not used in the method, but still it would be better to be explicit here by passing `None` instead of `evt`.